flake.nix

```diff
@@ -3,6 +3,9 @@
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs";
 
+    # flake-lang.nix
+    flake-lang.url = "github:mlabs-haskell/flake-lang.nix/jared/add-overlay-to-flake-typescript";
```

Comment:
Ugh, overlays are ugly, why did you end up using that?

Reply:
Oh goodness, this got somewhat convoluted (somewhat related PR [over here in flake-lang.nix](https://github.com/mlabs-haskell/flake-lang.nix/pull/3)

The overlay was used here:

- https://github.com/mlabs-haskell/lambda-buffers/blob/ac244a70e74d615abefd641e124ebaa4370e72eb/extras/lbf-nix/lbf-typescript.nix#L259-L282

where

1. a .lbf file was compiled (way above in the file)
2. we create a typescript project (but it's missing the extra dependencies from nix) in variable `lbTypescriptSrc`
3. so, when we create the `typescriptFlake`, we take the typescript project in the previous step, but we overlay its `postConfigure` phase to actually install the extra dependencies from nix.

Why didn't I add the extra dependencies in 2.? I suppose I was thinking code reuse so I wouldn't have to build the folder with all the extra dependencies once in 2.; and once again in 3. (3. will always do this automatically).

Perhaps there's a better way :^). Always open to suggestions :).

Reply:
No strong opinions, I trust you